machines/VCItemMachine/ExistingMosipVCItem/ExistingMosipVCItemMachine.ts

```diff
@@ -413,18 +389,61 @@ export const ExistingMosipVCItemMachine =
             src: 'verifyCredential',
             onDone: [
               {
-                actions: ['markVcValid', 'storeContext', 'updateVc'],
-                target: 'idle',
+                actions: ['setVerifiableCredential', 'storeContext'],
               },
             ],
             onError: [
               {
-                actions: log((_, event) => (event.data as Error).message),
-                target: 'idle',
+                //To-Do Handle Error Scenarios
+                actions: ['updateVerificationErrorMessage'],
+                target: 'handleVCVerificationFailure',
               },
             ],
           },
+          initial: 'idle',
+          on: {
+            STORE_RESPONSE: {
+              actions: [
+                'updateVc',
+                'logDownloaded',
+                'sendTelemetryEvents',
+                'removeVcFromInProgressDownloads',
+              ],
+              target: '.triggerAutoBackupForVcDownload',
+            },
+            STORE_ERROR: {
+              target: '#vc-item.checkingServerData.savingFailed',
+            },
+          },
+          states: {
+            idle: {},
+            triggerAutoBackupForVcDownload: {
+              invoke: {
+                src: 'isUserSignedAlready',
+                onDone: [
+                  {
+                    cond: 'isSignedIn',
+                    actions: ['sendBackupEvent'],
+                    target: '#vc-item.idle',
+                  },
+                  {
+                    target: '#vc-item.idle',
+                  },
+                ],
+              },
+            },
+          },
+        },
+
+        handleVCVerificationFailure: {
+          entry: ['removeVcMetaDataFromStorage'],
+          on: {
+            STORE_RESPONSE: {
+              actions: ['sendVerificationError'],
+            },
+          },
         },
+
         checkingVerificationStatus: {
           description:
             'Check if VC verification is still valid. VCs stored on the device must be re-checked once every [N] time has passed.',
```

Comment:
Hhat if the cond `isVCValid`(in line 452) is true ? How will we trigger autobackup for that scenario?

Reply:
We are not considering isVCValid() method now as its value is always false. We have it to handle other error scenario as part of INJI -695. Refer this comment(https://github.com/mosip/inji/pull/1228#discussion_r1479793926) . So only successful verification , we are storing the VC and trigger auto backup in the current flow.

Reply:
In case if it  has to return true in the future, please add a TODO for backup handling.

Reply:
Yeah sure.

Reply:
Added